model/riscv_zicfilp_regs.sail

```diff
@@ -0,0 +1,94 @@
+/*=======================================================================================*/
+/*  This Sail RISC-V architecture model, comprising all files and                        */
+/*  directories except where otherwise noted is subject the BSD                          */
+/*  two-clause license in the LICENSE file.                                              */
+/*                                                                                       */
+/*  SPDX-License-Identifier: BSD-2-Clause                                                */
+/*=======================================================================================*/
+
+function get_xLPE(p : Privilege) -> bool =
+  /* When S-mode is implemented:
+   *   Priv      xLPE
+   *   ----------------------
+   *   M         mseccfg.MLPE
+   *   S/HS      menvcfg.LPE
+   *   VS        henvcfg.LPE
+   *   U/VU      senvcfg.LPE
+   *
+   * When S-mode is not implemented:
+   *   Priv      xLPE
+   *   ----------------------
+   *   M         mseccfg.MLPE
+   *   U         menvcfg.LPE
+   */
+  match p {
+    Machine    => bool_bits(mseccfg[MLPE]),
+    Supervisor => bool_bits(menvcfg[LPE]),      // TODO: check henvcfg for VS mode
+    User       => if   currentlyEnabled(Ext_S)
+                  then bool_bits(senvcfg[LPE])
+                  else bool_bits(menvcfg[LPE]),
+  }
+
+function clause currentlyEnabled(Ext_Zicfilp) =
+  hartSupports(Ext_Zicfilp) & get_xLPE(cur_privilege)
+
+/* Architectural state for the Zicfilp extension. */
+
+register elp : bool
```

Comment:
Maybe add a comment that this stands for "expected landing pad"?
It's obvious once you read the code but having a comment makes it easier.